### ISSUE TITLE
how to create environment

### ISSUE THREAD

**Issue by mattiasospetti-medinnovations** [lllyasviel/ControlNet#739]
as title suggests: has any of you any idea on how to properly create a conda env for this repository?
The environment.yaml is outdated (some packages versions are from 2020).

**Comment by cowboy446** [comment]
Hey! I just had it run successfully in a conda env for stable-diffusion, whose PyTorch version is 2.8.0+cu128 (I manually configured instead of following outdated guides and other package versions in stable-diffusion's repo).
Two projects are alike since ControlNet is designed on stable-diffusion. Maybe you can try to deploy stable-diffusion first and directly `conda activate sd` and run ControlNet there.
By the way, I actually failed to launch controlnet's GUIs based on gradio(maybe that's my fault), so I rewrote scripts not in GUI loop but simply to interact in bash. That's how it worked.

**Comment by ZiQuanWW** [comment]
Hi there! I can run normally with this environment (there might be some unnecessary libraries), but the code needs to be modified. Torch is 2.4.1 cu121. Hope this helps you! 



Modified code:
Add:  , strict=False
model.load_state_dict(load_state_dict('./models/control_sd15_canny.pth', location='cuda')**, strict=False**)

Delete:  .queue()
block = gr.Blocks()【.queue()】<--delete this

[environment202511.yml](https://github.com/user-attachments/files/23724874/environment202511.yml)

environment.yaml：
channels:
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=5.1=1_gnu
  - ca-certificates=2025.11.4=h06a4308_0
  - ld_impl_linux-64=2.44=h153f514_2
  - libffi=3.4.4=h6a678d5_1
  - libgcc=15.2.0=h69a1729_7
  - libgcc-ng=15.2.0=h166f726_7
  - libgomp=15.2.0=h4751f2c_7
  - libstdcxx=15.2.0=h39759b7_7
  - libstdcxx-ng=15.2.0=hc03a8fd_7
  - libxcb=1.17.0=h9b100fa_0
  - libzlib=1.3.1=hb25bd0a_0
  - ncurses=6.5=h7934f7d_0
  - openssl=3.0.18=hd6dcaed_0
  - pthread-stubs=0.3=h0ce48e5_1
  - python=3.8.18=h955ad1f_0
  - readline=8.3=hc2a1206_0
  - setuptools=75.1.0=py38h06a4308_0
  - sqlite=3.51.0=h2a70700_0
  - tk=8.6.15=h54e0aa7_0
  - wheel=0.44.0=py38h06a4308_0
  - xorg-libx11=1.8.12=h9b100fa_1
  - xorg-libxau=1.0.12=h9b100fa_0
  - xorg-libxdmcp=1.1.5=h9b100fa_0
  - xorg-xorgproto=2024.1=h5eee18b_1
  - xz=5.6.4=h5eee18b_1
  - zlib=1.3.1=hb25bd0a_0
  - pip:
      - absl-py==2.3.1
      - addict==2.4.0
      - aiofiles==23.2.1
      - aiohappyeyeballs==2.4.4
      - aiohttp==3.10.11
      - aiosignal==1.3.1
      - albumentations==1.3.1
      - altair==5.1.2
      - annotated-doc==0.0.4
      - annotated-types==0.7.0
      - antlr4-python3-runtime==4.9.3
      - anyio==4.6.2
      - async-timeout==5.0.1
      - attrs==25.3.0
      - backports-zoneinfo==0.2.1
      - basicsr==1.4.2
      - blinker==1.8.2
      - braceexpand==0.1.7
      - cachetools==5.5.2
      - certifi==2025.11.12
      - charset-normalizer==3.4.4
      - click==8.1.8
      - contourpy==1.1.1
      - cycler==0.12.1
      - einops==0.7.0
      - exceptiongroup==1.3.1
      - fastapi==0.121.3
      - ffmpy==0.5.0
      - filelock==3.13.1
      - fonttools==4.57.0
      - frozenlist==1.5.0
      - fsspec==2023.9.2
      - ftfy==6.2.3
      - future==1.0.0
      - gitdb==4.0.12
      - gitpython==3.1.45
      - google-auth==2.43.0
      - google-auth-oauthlib==1.0.0
      - gradio==3.38.0
      - gradio-client==0.7.1
      - grpcio==1.70.0
      - h11==0.16.0
      - httpcore==1.0.9
      - httpx==0.28.1
      - huggingface-hub==0.17.3
      - idna==3.11
      - imageio==2.31.6
      - imageio-ffmpeg==0.4.9
      - importlib-metadata==6.11.0
      - importlib-resources==6.4.5
      - invisible-watermark==0.2.0
      - jinja2==3.1.6
      - joblib==1.4.2
      - jsonschema==4.23.0
      - jsonschema-specifications==2023.12.1
      - kiwisolver==1.4.7
      - kornia==0.7.1
      - lazy-loader==0.4
      - lightning-utilities==0.11.9
      - linkify-it-py==2.0.3
      - lmdb==1.7.5
      - markdown==3.7
      - markdown-it-py==2.2.0
      - markupsafe==2.1.5
      - matplotlib==3.7.5
      - mdit-py-plugins==0.3.3
      - mdurl==0.1.2
      - mpmath==1.3.0
      - multidict==6.1.0
      - networkx==3.0
      - numpy==1.24.4
      - nvidia-cublas-cu12==12.1.3.1
      - nvidia-cuda-cupti-cu12==12.1.105
      - nvidia-cuda-nvrtc-cu12==12.1.105
      - nvidia-cuda-runtime-cu12==12.1.105
      - nvidia-cudnn-cu12==9.1.0.70
      - nvidia-cufft-cu12==11.0.2.54
      - nvidia-curand-cu12==10.3.2.106
      - nvidia-cusolver-cu12==11.4.5.107
      - nvidia-cusparse-cu12==12.1.0.106
      - nvidia-nccl-cu12==2.20.5
      - nvidia-nvjitlink-cu12==12.9.86
      - nvidia-nvtx-cu12==12.1.105
      - oauthlib==3.3.1
      - omegaconf==2.3.0
      - open-clip-torch==2.24.0
      - opencv-python==4.8.1.78
      - opencv-python-headless==4.12.0.88
      - orjson==3.10.15
      - packaging==23.2
      - pandas==2.0.3
      - pillow==10.0.1
      - pip==25.0.1
      - pkgutil-resolve-name==1.3.10
      - platformdirs==4.3.6
      - prettytable==3.9.0
      - propcache==0.2.0
      - protobuf==4.25.8
      - pyarrow==17.0.0
      - pyasn1==0.6.1
      - pyasn1-modules==0.4.2
      - pydantic==2.9.2
      - pydantic-core==2.23.4
      - pydeck==0.9.1
      - pydub==0.25.1
      - pygments==2.19.2
      - pyparsing==3.1.4
      - python-dateutil==2.9.0.post0
      - python-multipart==0.0.20
      - pytorch-lightning==1.9.4
      - pytz==2025.2
      - pywavelets==1.4.1
      - pyyaml==6.0.3
      - qudida==0.0.4
      - referencing==0.35.1
      - regex==2024.11.6
      - requests==2.32.4
      - requests-oauthlib==2.0.0
      - rich==13.9.4
      - rpds-py==0.20.1
      - rsa==4.9.1
      - safetensors==0.4.1
      - scikit-image==0.21.0
      - scikit-learn==1.3.2
      - scipy==1.10.1
      - semantic-version==2.10.0
      - sentencepiece==0.2.0
      - six==1.17.0
      - smmap==5.0.2
      - sniffio==1.3.1
      - starlette==0.44.0
      - streamlit==1.28.0
      - streamlit-drawable-canvas==0.8.0
      - sympy==1.13.3
      - tb-nightly==2.14.0a20230808
      - tenacity==8.5.0
      - tensorboard==2.14.0
      - tensorboard-data-server==0.7.2
      - test-tube==0.7.5
      - threadpoolctl==3.5.0
      - tifffile==2023.7.10
      - timm==0.9.12
      - tokenizers==0.14.1
      - toml==0.10.2
      - tomli==2.3.0
      - toolz==1.0.0
      - torch==2.4.1+cu121
      - torchaudio==2.4.1+cu121
      - torchmetrics==1.1.2
      - torchvision==0.19.1+cu121
      - tornado==6.4.2
      - tqdm==4.67.1
      - transformers==4.34.0
      - triton==3.0.0
      - typing-extensions==4.8.0
      - tzdata==2025.2
      - tzlocal==5.2
      - uc-micro-py==1.0.3
      - urllib3==2.2.3
      - uvicorn==0.33.0
      - validators==0.34.0
      - watchdog==4.0.2
      - wcwidth==0.2.14
      - webdataset==0.2.85
      - websockets==11.0.3
      - werkzeug==3.0.6
      - yapf==0.40.1
      - yarl==1.15.2
      - zipp==3.20.2
